### ISSUE TITLE
SHM-2024: Task: Investigate partner issue with NFT app by using our in development TX playback technology

### ISSUE THREAD

**Issue by chrischabot** [shardeum-mirror/shardeum#9]
field | value
| - | - |
url | https://linear.app/shm/issue/SHM-2024/task-investigate-partner-issue-with-nft-app-by-using-our-in
author | @
date | Fri Aug  4 03:29:37 BST 2023
state | Canceled
project | Ops Strike Team
priority | High
assignee | @
labels | Strike Team
related | 
parent | 
children | 
PRs | 
attachments | 

**Here is the original problem post:**

Sandipan — Today at 7:36 AM One of our partner reported inconsistent behaviour in transactions:

these transaction [https://explorer-dapps.shardeum.org/tx/0x747b8af3a1e36a8bf3c75d1f6bd4399ab0083fa14439c8b66a3433c9f0a8716b](https://explorer-dapps.shardeum.org/tx/0x747b8af3a1e36a8bf3c75d1f6bd4399ab0083fa14439c8b66a3433c9f0a8716b) and [https://explorer-dapps.shardeum.org/tx/0x5f9d480b48d9e4812b23d9897427ca11157b59bddb887a9a2be5e9e491eb9d36](https://explorer-dapps.shardeum.org/tx/0x5f9d480b48d9e4812b23d9897427ca11157b59bddb887a9a2be5e9e491eb9d36) both has same data passed to the contract, one has minted the NFT and other one is a blank txn(has consumed more gas). This is happening at random intervals contract address: [https://explorer-dapps.shardeum.org/account/0x6fecf83c7ba12ce4a293b0f27e9b5b784b7f0099](https://explorer-dapps.shardeum.org/account/0x6fecf83c7ba12ce4a293b0f27e9b5b784b7f0099), if you observe the txns all are invoking the same function 0xb6ba3fcd, even though all txns are passed only few has minted the NFTs.

cc. @marc.hanson If this is actually network related, this can be high priority. Will be great if someone can check this out.

//////UPDATE:

Sandipan — Today at 9:41 AM Thanks a lot! And regarding the question you asked about conditions, They mentioned:

" No @Sandipan Kundu on success txn, the NFT should be minted or the txn should revert. One thing to observe here is that this is happening at random interval, if you try 5 times you might get the NFT 2 times provided same transaction and no time control logics. "

# To-do

* [ ] Use the in development playback tech and manual or semi manual efforts to build playback data sets so we can replay one good and one bad TX.
* [ ] report back when these data sets are ready and the TXs can be played back.
* [ ] if changes are needed to the playback tech feel free to make those as well, (but please just give status updates on this)
* [ ] once we can playback these TX we can include more people to help with the EVM analysis. I have added @Thant to this to help with that
* [ ] If we can identify a problem/potential fix then we can spawn a new task for that. Or if it is something the partner needs to fix we can work with @Sandipan Kundu to inform them

**Comment by chrischabot** [comment]
field | value
|-|-|
url | https://linear.app/shm/issue/SHM-2024/task-investigate-partner-issue-with-nft-app-by-using-our-in#comment-f62acea5
author | @
date | Fri Aug  4 03:32:52 BST 2023

Thant 6/21/23: @Sandipan Kundu it is hard to figure out just by looking at the receipt. It will be helpful if we can have a glance of, at least, the function theses txs are calling. My guess is it may be using an opcode which has inderministic nature or may be opcode shardeum VM does not support.

**Comment by chrischabot** [comment]
field | value
|-|-|
url | https://linear.app/shm/issue/SHM-2024/task-investigate-partner-issue-with-nft-app-by-using-our-in#comment-5b4c4305
author | @
date | Fri Aug  4 03:33:40 BST 2023

mhanson 6/21/23: @Thant you need input from @Sandipan Kundu to proceed?

**Comment by chrischabot** [comment]
field | value
|-|-|
url | https://linear.app/shm/issue/SHM-2024/task-investigate-partner-issue-with-nft-app-by-using-our-in#comment-df05796d
author | @
date | Fri Aug  4 03:34:00 BST 2023

Thant 6/21/23: @mhanson @Sandipan Kundu I have tested ERC 721A minting here. [https://discord.com/channels/448919918063124480/1071190531369083011/1121112667327828028](https://discord.com/channels/448919918063124480/1071190531369083011/1121112667327828028). We cannot proceed the investigation unless we know the full contract codes or we have finished in-development tx playback too.

**Comment by chrischabot** [comment]
field | value
|-|-|
url | https://linear.app/shm/issue/SHM-2024/task-investigate-partner-issue-with-nft-app-by-using-our-in#comment-5640e6d3
author | @
date | Fri Aug  4 03:34:27 BST 2023

Sandipan Kundu 6/21/23: @Thant I have asked for the entire smart contract, let’s see if they can share that. Doe we have an ETA for the playback tech if not?

**Comment by chrischabot** [comment]
field | value
|-|-|
url | https://linear.app/shm/issue/SHM-2024/task-investigate-partner-issue-with-nft-app-by-using-our-in#comment-49c605d3
author | @
date | Fri Aug  4 03:35:00 BST 2023

Atharva Nimbalkar 6/22/23: The Playback tech should be partially ready by the 25th of this month. It should be enough to run handpicked TXs and give us significantly helpful insights.

**Comment by chrischabot** [comment]
field | value
|-|-|
url | https://linear.app/shm/issue/SHM-2024/task-investigate-partner-issue-with-nft-app-by-using-our-in#comment-4e43958e
author | @
date | Fri Aug  4 03:35:38 BST 2023

Atharva Nimbalkar 6/22/23: I’ll keep providing regular updates on the playback tech in the description of this ticket.

**Comment by chrischabot** [comment]
field | value
|-|-|
url | https://linear.app/shm/issue/SHM-2024/task-investigate-partner-issue-with-nft-app-by-using-our-in#comment-861592a2
author | @
date | Fri Aug  4 03:36:00 BST 2023

Atharva Nimbalkar 6/22/23: Discussed with @mhanson, setting the assignee to me.

**Comment by chrischabot** [comment]
field | value
|-|-|
url | https://linear.app/shm/issue/SHM-2024/task-investigate-partner-issue-with-nft-app-by-using-our-in#comment-7531749c
author | @
date | Fri Aug  4 03:36:40 BST 2023

Sandipan Kundu 6/22/23: @Thant Here are the smart contracts: [https://gist.github.com/BanditArul/ea73baea323f6365972d9f22a1e7f354](https://gist.github.com/BanditArul/ea73baea323f6365972d9f22a1e7f354)

**Comment by chrischabot** [comment]
field | value
|-|-|
url | https://linear.app/shm/issue/SHM-2024/task-investigate-partner-issue-with-nft-app-by-using-our-in#comment-967fbbcf
author | @
date | Thu Aug 10 13:46:30 BST 2023

Added this to "backlog" because we need new clue from the partner team.
